src/gasx.f90

```diff
@@ -704,7 +707,7 @@ SUBROUTINE o2flux(T, S, kw660, ppo, o2, dz1, N, o2ex)
 
   DO i = 1, N
 !     Transfer velocity for O2 in m/s [4]
-      kwo2 = (kw660(i) * (660/sco2(T))**0.5)
+      kwo2 = (kw660(i) * (660/sco2(T(i)))**0.5)
```

Comment:
This was complaining about a rank mismatch. Not sure if the loop is supposed to go over T as well.